### PR TITLE
serialfile: localize os.Open into NewSerialFile

### PR DESCRIPTION
**Pull request by rht**

### CODE REVIEW

File: commands/files/serialfile.go [hunk 1]
```diff
@@ -42,21 +36,11 @@ func newSerialFile(name, path string, file *os.File, stat os.FileInfo) (File, er
 
 	// for directories, stat all of the contents first, so we know what files to
 	// open when NextFile() is called
-	contents, err := file.Readdir(0)
+	contents, err := ioutil.ReadDir(path)
```
Comment (whyrusleeping) [inline commands/files/serialfile.go]: why change this? we already have the file. no need to make an extra syscall

File: commands/files/serialfile.go [hunk 2]
```diff
@@ -42,21 +36,11 @@ func newSerialFile(name, path string, file *os.File, stat os.FileInfo) (File, er
...
 
-	// we no longer need our root directory file (we already statted the contents),
-	// so close it
-	err = file.Close()
-	if err != nil {
-		return nil, err
-	}
-
-	// make sure contents are sorted so -- repeatably -- we get the same inputs.
-	sort.Sort(sortFIByName(contents))
```
Comment (whyrusleeping) [inline commands/files/serialfile.go]: why remove this?

Comment (whyrusleeping) [inline commands/files/serialfile.go]: I guess it doesnt matter for building a dag, since links are sorted anyways.

Comment (rht) [inline commands/files/serialfile.go]: `ioutil.ReadDir` already does the sorting

File: commands/files/serialfile.go (hunk 1 above)
Comment (rht) [inline commands/files/serialfile.go]: What if `os.Open()` is done only when `!stat.IsDir()`?

Comment (rht) [inline commands/files/serialfile.go]: Not possible because in `NextFile()`, `f.Close()` requires the opened `os.File` object.

Comment (rht) [inline commands/files/serialfile.go]: Still possible but then `NextFile()` no longer uses `newSerialFile()`.

File: commands/files/serialfile.go (hunk 2 above)
Comment (rht) [inline commands/files/serialfile.go]: Reverted `ioutil.ReadDir`

**Comment by jbenet** [comment]
LGTM. @whyrusleeping ?

**Comment by jbenet** [comment]
I think this needs rebasing on top of symlink PR

**Comment by jbenet** [comment]
I think this needs rebasing over symlinks PR

**Comment by rht** [comment]
Rebased

**Comment by whyrusleeping** [comment]
that goprocess issue is happening more and more frequently. Otherwise, this LGTM

**Comment by rht** [comment]
@jbenet this is one of #1641's dependency. Rebased.

**Pull request merged**